src/Emulator/EmulatorProvider.tsx

```diff
@@ -36,11 +36,11 @@ const DialogueContext = createContext<DialogueInterface>({
 });
 
 const DialogueProvider: FC = ({ children }) => {
-  const [dialogueMessages, setDialogueMessages] = useState<string[]>([]);
+  const [dialogueMessages, setDialogueMessages] = useState<string[][]>([]);
 
   // Not using useCallback so we can add the same error messages repeatedly
-  const appendDialogue = (message: string) => {
-    setDialogueMessages(dialogueMessages.concat([message]));
+  const appendDialogue = (message: string, color = 'white') => {
+    setDialogueMessages(dialogueMessages.concat([[message, color]]));
```

Comment:
Then this would look like

```ts
setDialogueMessages(dialogueMessages.concat([{ message, color }]))
```

Reply:
Makes total sense, I should have considered extensibility more.

Making the changes now.